javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -30,15 +34,40 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class TracerInstaller {
-  private static final Logger log = LoggerFactory.getLogger(TracerInstaller.class);
+@AutoService(ComponentInstaller.class)
+public class OpenTelemetryInstaller implements ComponentInstaller {
+  private static final Logger log = LoggerFactory.getLogger(OpenTelemetryInstaller.class);
 
   private static final String EXPORTER_JAR_CONFIG = "otel.exporter.jar";
   private static final String EXPORTERS_CONFIG = "otel.exporter"; // this name is from spec
   private static final String PROPAGATORS_CONFIG = "otel.propagators"; // this name is from spec
   private static final String JAVAAGENT_ENABLED_CONFIG = "otel.javaagent.enabled";
   private static final List<String> DEFAULT_EXPORTERS = Collections.singletonList("otlp");
 
+  @Override
+  public void beforeByteBuddyAgent() {
+    // calling (shaded) OpenTelemetry.getGlobalTracerProvider() with context class loader set to
+    // the
+    // agent class loader, so that SPI finds the agent's (isolated) SDK, and (shaded)
+    // OpenTelemetry registers it, and then when instrumentation calls (shaded)
+    // OpenTelemetry.getGlobalTracerProvider() later, they get back the agent's (isolated) SDK
+    //
+    // but if we don't trigger this early registration, then if instrumentation is the first to
+    // call (shaded) OpenTelemetry.getGlobalTracerProvider(), then SPI can't see the agent class
+    // loader,
+    // and so (shaded) OpenTelemetry registers the no-op TracerFactory, and it cannot be replaced
+    // later
```

Comment:
I'm not sure this comment makes sense here

Reply:
where to put it, @mateuszrzeszutek suggested to put it here.

Reply:
ya, this is a good place, sorry about that 👍